scripts/tablegen.py

```diff
@@ -36,12 +37,43 @@
 # comments are needed in order to decode the commented helm
 # values.yaml file
 #
-ISTIO_CONFIG_DIR = "istio/install/kubernetes/helm/istio"
+ISTIO_CONFIG_DIR = "install/kubernetes/helm/istio"
 YAML_CONFIG_DIR = ISTIO_CONFIG_DIR + "/charts"
 VALUES_YAML = "values.yaml"
-ISTIO_IO_DIR = os.path.abspath(__file__ + "/../../")
 CONFIG_INDEX_DIR = "content/en/docs/reference/config/installation-options/index.md"
-
+ISTIO_REPO = "istio-repo"
+
+def downloadIstioRepo(currentRelease):
+    releaseName = "release-" + currentRelease
+    curl_command = "git clone --depth=1 -q -b %s 'https://github.com/istio/istio.git' %s"
+    status = os.system(curl_command %(releaseName, ISTIO_REPO))
+    if status != 0:
+        print("An error occured trying to clone Istio repo for release: %s." %releaseName)
+        exit()
+
+def deleteIstioRepo():
+    os.system("rm -rf %s" %ISTIO_REPO)
+
+def getCurrentReleaseTag():
```

Comment:
I think this function needs to be replaced with a hardcoded constants for the release identity. Otherwise, when we create a release-1.3 branch of istio.io and try to update the reference docs there, it'll pull the content from the latest release in istio/istio instead of being pinned to the release-1.3 branch.

Basically, we need to have explicit control in istio/istio.io of which branch of istio/istio we pull stuff from. This is why we hardcode the branch names in scripts/grab_reference_docs.sh and I think we need to do the same here.

Reply:
@geeknoid I was trying to automatically get the latest release number for Istio by using the release tag API request: api.github.com/repos/istio/istio/tags gives us the list of tags for istio and from this I try to get the highest value. So for example, from a list of  1.3.0-rc.0 , 1.2.0-rc.0 , 1.1.0-rc.0, etc pick 1.3.0-rc.0 and use 1.3 as the release: git clone --depth=1 -q -b istio-repo 'https://github.com/istio/istio.git' release-1.3. 

I can go ahead and hardcode it if this is not the right thing to do. I was trying to see if there was a way we could avoid hardcoding values. Thanks @geeknoid, appreciate your feedback.

Reply:
Yeah, I saw that you were figuring out the version automatically, but the problem is basically that "latest" is often not the right choice. Within the release-1.3 branch of istio/istio.io, we want to get the reference state for the release-1.3 branch of istio/istio and not the latest from master.

So I think pinning to specific release versions is necessary.

Reply:
Fixed it so that it we now have a hardcoded release of Istio. Thanks @geeknoid.